### PR TITLE
[SDP-1245] Invalidate Circle Distribution Account Status upon receiving auth error

### PR DESCRIPTION
**Pull request by ziyliu**
### What

We should mark Circle distribution accounts whose API keys are not either expired (401) or lack transfer request permissions through Circle (403) as "deactivated". Requests to SDP to process payments will not trigger any transfer request to Circle that will determinately fail.

### Why

[TODO: Why this change is being made. Include any context required to understand the why.]

### Known limitations

[TODO or N/A]

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [x] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: stellar-multitenant/pkg/tenant/manager_test.go [hunk 1]
```diff
@@ -363,31 +363,31 @@ func Test_Manager_GetTenantByIDOrName(t *testing.T) {
...
 	})
 
 	t.Run("returns error when tenant is not found", func(t *testing.T) {
-		tntDB, err := m.GetTenantByIDOrName(ctx, "unknown")
-		assert.ErrorIs(t, err, ErrTenantDoesNotExist)
+		tntDB, dbErr := m.GetTenantByIDOrName(ctx, "unknown")
+		assert.ErrorIs(t, dbErr, ErrTenantDoesNotExist)
 		assert.Nil(t, tntDB)
 	})
 }
```
Comment (marcelosalloum) [inline stellar-multitenant/pkg/tenant/manager_test.go:393]: 🤔  Were these changes needed due to shadowing issues, or did you change it because you preferred this new name for the error variable?

We don't usually give custom names to error variables unless they are a specific error type, or to prevent variable shadowing.

File: stellar-multitenant/pkg/tenant/manager_test.go [hunk 2]
```diff
@@ -570,29 +570,86 @@ func Test_Manager_SoftDeleteTenantByID(t *testing.T) {
...
+				Status:                  pointerTo(ActivatedTenantStatus),
+			})
+		require.NoError(t, err)
+		err = m.DeactivateTenantDistributionAccount(ctx, tnt.ID)
+		require.NoError(t, err)
+
+		dbTnt, dbErr := m.GetTenant(ctx, &QueryParams{Filters: map[FilterKey]interface{}{FilterKeyID: tnt.ID}})
+		require.NoError(t, dbErr)
+		assert.Equal(t, schema.AccountStatusPendingUserActivation, dbTnt.DistributionAccountStatus)
+	})
```
Comment (marcelosalloum) [inline stellar-multitenant/pkg/tenant/manager_test.go:650]: 👋  could you include a test case to validate that this method won't throw an error if the distribution account status is already `PENDING_USER_ACTIVATION?

File: internal/circle/client.go [hunk 3]
```diff
@@ -192,4 +195,25 @@ func (client *Client) request(ctx context.Context, u string, method string, isAu
...
+			return fmt.Errorf("deactivating tenant distribution account: %w", deactivateTntErr)
+		}
+	}
+
+	apiError, err := parseAPIError(resp)
+	if err != nil {
+		return fmt.Errorf("parsing API error: %w", err)
+	}
+
+	return fmt.Errorf("API error: %w", apiError)
```
Comment (marcelosalloum) [inline internal/circle/client.go:216]: 💡 
```suggestion
	return fmt.Errorf("Circle API error: %w", apiError)
```

File: internal/circle/client.go [hunk 4]
```diff
@@ -192,4 +195,25 @@ func (client *Client) request(ctx context.Context, u string, method string, isAu
...
+}
+
```
Comment (marcelosalloum) [inline internal/circle/client.go:218]: 👋 I'm missing unit tests for this function here. Could you please add that?

**Review by marcelosalloum — APPROVED** [review]
LGTM! Plz, feel free to merge it after addressing the comments below.

In special, the one I marked with 👋 is the most important change.

File: stellar-multitenant/pkg/tenant/manager_test.go (hunk 1 above)
Comment (ziyliu) [reply stellar-multitenant/pkg/tenant/manager_test.go:393]: yes exactly, I had to change them due to shadowing issues that the linter didn't like!
